pkg/asset/installconfig/aws/validation.go

```diff
@@ -329,12 +330,28 @@ func validateEndpointAccessibility(endpointURL string) error {
 	if port == "" {
 		port = "https"
 	}
-	conn, err := proxy.Dial(context.Background(), "tcp", net.JoinHostPort(URL.Hostname(), port))
-	if err != nil {
-		return err
+	err = proxyDialURL(context.Background(), net.JoinHostPort(URL.Hostname(), port))
+	return err
+}
+
+func proxyDialURL(ctx context.Context, URL string) error {
+	conn, err := proxy.Dial(context.Background(), "tcp", URL)
```

Comment:
So this satisfies the case if the proxy is setup using `ALL_PROXY`? If that is correct, can you add a comment indicating that is the reason we are using this?

Also this seems like the less common use case, so I would move this to be the second attempt